? config.guess
? config.sub
? gnome-mount-0.4-0.cvs20060213.tar.gz
? gnome-mount-0.4-drive-mount.patch
? gnome-mount-fix-segv-on-no-media.patch
? gnome-mount-nicknames.patch
? libtool
? ltmain.sh
? po/Makefile
? po/Makefile.in
? po/Makefile.in.in
? po/POTFILES
? po/it.gmo
? po/ja.gmo
? po/nb.gmo
? po/stamp-it
Index: src/gnome-mount.c
===================================================================
RCS file: /cvs/gnome/gnome-mount/src/gnome-mount.c,v
retrieving revision 1.17
diff -u -p -r1.17 gnome-mount.c
--- src/gnome-mount.c	25 Feb 2006 00:39:31 -0000	1.17
+++ src/gnome-mount.c	1 Mar 2006 16:00:21 -0000
@@ -238,7 +238,8 @@ show_error_dialog_unmount (const char *u
 /* #define MOUNT_ERROR_DIALOG_RESPONSE_INSTALL_DRIVER 10 see below */
 
 static void
-show_error_dialog_mount (LibHalVolume *volume, const char *error_name, const char *error_detail, 
+show_error_dialog_mount (LibHalVolume *volume, LibHalDrive *drive,
+			 const char *error_name, const char *error_detail, 
 			 const char *fstype_requested)
 {
 	GtkWidget *w;
@@ -266,7 +267,10 @@ show_error_dialog_mount (LibHalVolume *v
 					    0,
 					    _("Cannot mount volume"));
 
-		volume_name = libhal_volume_get_label (volume);
+		if (volume != NULL)
+			volume_name = libhal_volume_get_label (volume);
+		else
+			volume_name = NULL;
 
 		if (strcmp (error_name, "org.freedesktop.Hal.Device.Volume.PermissionDenied") == 0) {
 			gtk_message_dialog_format_secondary_text (
@@ -289,8 +293,8 @@ show_error_dialog_mount (LibHalVolume *v
 				volume_name != NULL ? 
 				_("The volume '%s' uses the <i>%s</i> file system which is not supported by your system.") :
 				_("The volume uses the <i>%s</i> file system which is not supported by your system."),
-				volume_name != NULL ? volume_name : fstype_requested,
-				volume_name != NULL ? fstype_requested : "");
+				volume_name != NULL ? volume_name : (fstype_requested != NULL ? fstype_requested : ""),
+				volume_name != NULL ? (fstype_requested != NULL ? fstype_requested : "") : "");
 
 			/* some day.. :-)
 			gtk_dialog_add_buttons (GTK_DIALOG (w),
@@ -463,9 +467,9 @@ volume_mount_with_options (const char *u
 
 		notify_parent (FALSE);
 
-		show_error_dialog_mount (volume, error.name, error.message, 
+		show_error_dialog_mount (volume, drive, error.name, error.message, 
 					 (fstype != NULL && strlen (fstype) > 0) ? 
-					 fstype : libhal_volume_get_fstype (volume));
+					 fstype : (volume != NULL ? libhal_volume_get_fstype (volume) : NULL));
 
 		dbus_error_free (&error);
 		goto out;
@@ -1701,7 +1705,7 @@ try_drive:
 
 out:	
  	if (drive != NULL)
-		libhal_volume_free (volume);
+		libhal_drive_free (drive);
 
  	if (volume != NULL)
 		libhal_volume_free (volume);
